_includes/manuals/1.24/5.7.7_openid_sso.md

```diff
@@ -0,0 +1,106 @@
+### Single Sign-on for Foreman using OpenID Connect protocol
+
+An OpenID Provider(OP)  implements a Single Sign-on (SSO) using  an OpenID Connect (OIDC) protocol. OIDC verifies the identity of the End-user depending on the given authentication details. In this example, [Keycloak](https://www.keycloak.org/) is used as the OpenID provider, and Foreman is the Relying Party (RP)
+
+OpenID provider provides an ID Token, encoded in JSON Web Token (JWT) to Foreman.
+
+An example of an encoded JWT ID Token:
+
+```
+eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c
+```
+
+#### Configuring Foreman
+
+As a system administrator, execute the following procedure to configure Foreman.
+
+Prerequisites
+
+* A working installation of Foreman at https://foreman.example.com.
+* A working installation of OpenID provider, for example, Keycloak at https://sso.example.com, which uses OIDC protocol. Ensure that the Keycloak server is running.
+* Ensure that both RP and OpenID provider are using https instead of http. If the certificates or the CA is self-signed, ensure that they are added to the End-user certificate trust store.
+
+Procedure
+
+1. Install the mod_auth packages:
+```
+# yum install mod_auth_openidc python2-requests-oauthlib keycloak-httpd-client-install -y
+```
+
+2. Register Katello/Foreman as application in the Keycloak:
+```
+# export KEYCLOAK_URL=https://keycloak.example.com
+# export KEYCLOAK_REALM=ssl-realm
+# export KEYCLOAK_USER=admin
+# keycloak-httpd-client-install --app-name foreman-openidc --keycloak-server-url $KEYCLOAK_URL --keycloak-admin-username $KEYCLOAK_USER --keycloak-realm $KEYCLOAK_REALM --keycloak-admin-realm master --keycloak-auth-role root-admin -t openidc -l /users/extlogin -d
+```
+Warning: This feature is not yet supported by foreman-installer. As a result, re-running the foreman-installer command can delete the changes in Apache files added by this feature.
+
+3. Log in to the Foreman admin page and click **Administer** → **Settings** → **Authentication**.
+ ![Foreman UI](Foreman_UI.png)
+
+4. Set **authorize_login_delegation_auth_source_user_autocreate** to **External** to enable auto-creation of users from external OpenID provider.
+
+5. Set the following OIDC parameters:
+
+ **OIDC Algorithm**: Algorithm type with which JWT was encoded in the OpenID provider. For example, For example, set this parameter to RS-256.
+
+ **OIDC Audience**: Name of the OIDC audience used for authentication. In case of [Keycloak](https://www.keycloak.org/), it is Client ID.
```

Comment:
Do we need to link to Keycloak project's website again? We've already provided its link in the introduction at the top. What convention do other manuals follow?